pebblo/app/service/loader/loader_doc_service.py

```diff
@@ -0,0 +1,239 @@
+                    name=loader_name,
+                    sourcePath=source_path,
+                    sourceType=source_type,
+                    sourceSize=source_size,
+                    sourceFiles=loader_source_files,
+                    lastModified=get_current_time(),
+                )
+                loader_list.append(new_loader_data.dict())
+                app_loader_details["loaders"] = loader_list
+
+        # self.db.update_data(table_obj, app_loader_details)
+        logger.debug("Loader details Updated successfully.")
+        return app_loader_details
+
+    @timeit
+    def _get_doc_classification(self, doc):
+        logger.debug("Doc classification started.")
+        doc_info = AiDataModel(
+            data=doc.get("doc", None),
+            entities={},
+            entityCount=0,
+            topics={},
+            topicCount=0,
+        )
+        try:
+            if doc_info.data:
+                topics, topic_count, topic_details = topic_classifier_obj.predict(
+                    doc_info.data
+                )
+                (
+                    entities,
+                    entity_count,
+                    anonymized_doc,
+                    entity_details,
+                ) = self.entity_classifier_obj.presidio_entity_classifier_and_anonymizer(
+                    doc_info.data,
+                    anonymize_snippets=ClassifierConstants.anonymize_snippets.value,
+                )
+                doc_info.topics = topics
+                doc_info.entities = entities
+                doc_info.topicCount = topic_count
+                doc_info.entityCount = entity_count
+                doc_info.data = anonymized_doc
+            logger.debug("Doc classification finished.")
+            return doc_info
+        except Exception as e:
+            logger.error(
+                f"Get Classifier Response Failed for doc: {doc}, Exception: {e}"
+            )
+            return doc_info
+
+    @staticmethod
+    @timeit
+    def _update_doc_details(doc, doc_info):
+        """
+        Create a doc model and return its object
+        """
+        logger.debug("Update doc details with classification result")
+        doc["entities"] = doc_info.entities
+        doc["topics"] = doc_info.topics
+        logger.debug("Input doc updated with classification result")
+
+    @timeit
+    def _doc_pre_processing(self):
+        logger.debug("Input docs pre processing started.")
+        input_doc_list = self.data.get("docs", [])
+        for doc in input_doc_list:
+            doc_info = self._get_doc_classification(doc)
+            self._update_doc_details(doc, doc_info)
+
+        # Update input doc with updated one
+        logger.debug("Doc pre processing finished.")
+
+    @timeit
+    def _get_or_create_data_source(self):
+        logger.debug("Getting or Creating Data Source Details.")
+        loader_details = self.data.get("loader_details") or {}
```

Comment:
Instead of self.data.get("loader_details") We can just do self.data.get("loader_details", {})

Reply:
This won't work in case when we get it as `{"loader_details": None}`.

Reply:
Even when loader-details is NONE it will return {} only from this statement